org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/javasupport/SimpleProjectDependencyGraph.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2019 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.builder.impl.javasupport;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.core.resources.IProject;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.inject.Singleton;
+
+/**
+ * @author Sebastian Zarnekow - Initial contribution and API
+ * 
+ * @since 2.19
+ */
+@Singleton
+public class SimpleProjectDependencyGraph {
+
+	private final Map<IProject, Set<IProject>> dependencies = Maps.newConcurrentMap();
```

Comment:
no bidimap intentional?

Reply:
Hmm that could simplify getDependentXtextProjects. Should be changed, yes.

Reply:
The impl of `#put` becomes harder if we make this bidi. Since we need getDependentXtextProjects only when the classpath changes, I'll keep it as is.